v-next/hardhat/test/internal/cli/init/template.ts

```diff
@@ -75,7 +75,7 @@ describe("template contents", async () => {
     return cleanup;
   }
 
-  const templates = await getTemplates();
+  const templates = await getTemplates("hardhat-3");
```

Comment:
Testing the hardhat 2 templates with this approach is complex and probably not worth it for a temporary solution. Instead, the templates were tested manually, as covered in the PR description.